third_party/terraform/resources/resource_composer_environment.go.erb

```diff
@@ -393,6 +397,44 @@ func resourceComposerEnvironment() *schema.Resource {
 								},
 							},
 						},
+<% end -%>
+<% unless version == "ga" -%>
+            "database_config": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							Computed:     true,
+							MaxItems:     1,
+							Description:  `The configuration of Cloud SQL instance that is used by the Apache Airflow software.`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"machine_type": {
+										Type:         schema.TypeString,
+										Optional:     true,
```

Comment:
Something we try to do is avoid having blocks with no `Required` fields. Setting it to `Required` only makes it required if a `database_config` block is in a user's config.  As written, this allows the following two configurations:

```
database_config {}
```

```
database_config {
  machine_type = "some-value"
}
```

However the first configuration is actually treated as a `nil` value by Terraform! In a confusing way, too- it's an `[]interface{}{nil}`. Most of the time we handle them correctly, but sometimes we don't and configs like that will crash a user. Making the `machine_type` sub-field `Required` means that users can only write the second config.

I think we can make both `machine_type` fields `Required`. If we end up with more fields in those blocks, we can relax the requirements (eg with `AtLeastOneOf`).

Reply:
Alternatively, you could add their default values (the "if not specified..." by setting a `Default` as well.